src/test/java/uk/gov/pay/api/it/directdebit/pact/DirectDebitPaymentTest.java

```diff
@@ -4,31 +4,32 @@
 import com.jayway.jsonassert.JsonAssert;
 import com.jayway.restassured.response.ValidatableResponse;
 import io.dropwizard.testing.junit.DropwizardAppRule;
-import java.time.ZonedDateTime;
-import javax.ws.rs.core.HttpHeaders;
 import org.apache.http.client.fluent.Executor;
 import org.junit.After;
 import org.junit.Rule;
 import org.junit.Test;
 import uk.gov.pay.api.app.PublicApi;
 import uk.gov.pay.api.app.config.PublicApiConfig;
-import uk.gov.pay.api.it.PaymentResourceITestBase;
 import uk.gov.pay.api.model.PaymentState;
 import uk.gov.pay.api.utils.ApiKeyGenerator;
 import uk.gov.pay.api.utils.DateTimeUtils;
-import uk.gov.pay.api.utils.JsonStringBuilder;
 import uk.gov.pay.commons.testing.pact.consumers.PactProviderRule;
 import uk.gov.pay.commons.testing.pact.consumers.Pacts;
 
+import javax.ws.rs.core.HttpHeaders;
+import java.time.ZonedDateTime;
+
 import static com.jayway.restassured.RestAssured.given;
 import static com.jayway.restassured.http.ContentType.JSON;
 import static io.dropwizard.testing.ConfigOverride.config;
 import static io.dropwizard.testing.ResourceHelpers.resourceFilePath;
 import static javax.ws.rs.core.HttpHeaders.AUTHORIZATION;
 import static org.hamcrest.CoreMatchers.nullValue;
 import static org.hamcrest.core.Is.is;
+import static uk.gov.pay.api.utils.Payloads.aSuccessfulPaymentPayload;
+import static uk.gov.pay.api.utils.Urls.directDebitFrontendSecureUrl;
 
-public class DirectDebitPaymentTest extends PaymentResourceITestBase {
```

Comment:
👍 Looks like the only thing we were taking from the base class was `paymentLocationFor()` which I see you've now added here. I'm in favour of a little duplication to make tests easier to understand.

Reply:
the reason I've removed the `extends` is because there are a bunch of things in PaymentResourceITestBase that aren't used here. i think the "favor composition over inheritance" principle Trumps "do not repeat yourself". i'm going to move the `paymentLocationFor` into the Urls class as you've suggested :)

Reply:
👍 Completely agree